Auxiliary/DirectXTexEXR.cpp

```diff
@@ -371,23 +371,35 @@ HRESULT DirectX::LoadFromEXRFile(const wchar_t* szFile, TexMetadata* metadata, S
 
         auto const dw = file.dataWindow();
 
-        const int width = dw.max.x - dw.min.x + 1;
-        const int height = dw.max.y - dw.min.y + 1;
+        int width = dw.max.x - dw.min.x + 1;
+        int height = dw.max.y - dw.min.y + 1;
+        size_t arraySize = 1;
 
         if (width < 1 || height < 1)
             return E_FAIL;
 
+        if (file.header().find("envmap") != file.header().end())
+        {
+            if (width == height / 6)
```

Comment:
This is a good fix, but I wonder if the `arraySize = 6;` should be in the `(width == height / 6)` case to just ignore the env map thing if the size isn't right.

Reply:
I.e:

```
if (width == height / 6)
{
    height = width;
    arraySize = 6;
}
```

Reply:
Maybe it will be better if the size isn't right to return `E_FAIL`. Per their documentation it says it should contain the flag `envmap`